### ISSUE TITLE
March 2021 Release

### ISSUE THREAD

**Issue by toasted-nutbread** [FooSoft/yomichan#1580]
https://github.com/FooSoft/yomichan/releases/tag/21.3.31.0

@FooSoft 

### General
* [x] Create [release tag](https://github.com/FooSoft/yomichan/tree/21.3.31.0).
* [x] Create [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.0).

### Testing
* [ ] Update Chrome testing release.
* [ ] Update Firefox testing release.
* [ ] Update Firefox [updates.json](https://github.com/FooSoft/foosoft.github.io/blob/master/projects/yomichan/dl/updates.json) and [verify](https://foosoft.net/projects/yomichan/dl/updates.json).
* [ ] Upload testing builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.0).

### Stable
* [ ] Update Chrome stable release.
* [ ] Update Firefox stable release.
* [ ] Upload stable builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.0).

**Comment by FooSoft** [comment]
Testing release up!

**Comment by Thermospore** [comment]
I'd like to try using the new option hotkey feature to toggle a dictionary. Is there an easy way to find the path for a given option? Might be nice to add a little explanation text in the menu there

Also does that only work for these kinds of options?
![image](https://user-images.githubusercontent.com/35350819/113380952-209cf980-932a-11eb-875a-2a77380a0bc0.png)
or can it cycle through dropdown lists as well

**Comment by toasted-nutbread** [comment]
Check [options-schema.json](https://github.com/FooSoft/yomichan/blob/master/ext/data/schemas/options-schema.json) or export your settings file and look at the paths. Paths start relative to profile[current].options, and the box will outline in red if the path is invalid. It only works on booleans.

**Comment by grospoulpe** [comment]
Just installed, with all the dictionaries I use (or not ;p), keeping the stable version (I suppose we have to, and eventually will become that new version)
So far, so good - and be able to play different audio source with a keyboard shortcut is great / timesaving :)
And the possibility to collapse dictionaries (especially with 研究社　新和英大辞典　第５版 , which has a lot of example displayed) is a useful feature, too :)

**Comment by Aridow** [comment]
I noticed it's no longer possible to drag and drop kanjis from kanjidics dictionaries on Firefox. Is it expected?
Here's what I mean on Chrome : 
![image](https://user-images.githubusercontent.com/65539338/113480892-06ccf500-9497-11eb-9844-df06fd2405f8.png)

**Comment by toasted-nutbread** [comment]
> I noticed it's no longer possible to drag and drop kanjis from kanjidics dictionaries on Firefox. Is it expected?

No, apparently this has been an issue for a while though. #1586 fixes it. If you want to patch your styles, you can add this custom CSS:
```css
.kanji-glyph {
    position: static;
    z-index: auto;
}
.kanji-tag-list {
  position: relative;
}
.entry[data-type=kanji] .entry-body-section[data-section-type=frequencies] {
  position: relative;
}
```

**Comment by qharscoet** [comment]
Hello,

I started diving into anki export customisation yesterday and saw that there are a lot of changes in this area in this release.
Would it be possible to expose back the JMDict sequence ID in the data ? Before it could be accessed in definitions.definitions[n].sequence but on testing it seems to be absent from the new anki logged format, or at -1. (see screenshort below)

(The Kanji Study app on Android allows dynamic open in app via a special URL that takes this as a parameter, really neat feature to be able to embed this in the cards when syncing with mobile) 

![image](https://user-images.githubusercontent.com/6672352/113487495-afd81780-94b8-11eb-9c92-abff3e77187e.png)

**Comment by toasted-nutbread** [comment]
> I started diving into anki export customisation yesterday and saw that there are a lot of changes in this area in this release.
> Would it be possible to expose back the JMDict sequence ID in the data ? Before it could be accessed in definitions.definitions[n].sequence but on testing it seems to be absent from the new anki logged format, or at -1.

Yes, I have added it back in: #1591. This will be included in a patch before the stable release is made.

**Comment by Thermospore** [comment]
Probably not new, I've seen similar cases before

If you search 殉じた, secondary search doesn't pull up the other 殉ずる definitions

![image](https://user-images.githubusercontent.com/35350819/113524392-d8b0e900-955a-11eb-8aed-aa6d79cb7011.png)

![image](https://user-images.githubusercontent.com/35350819/113524417-0433d380-955b-11eb-8bf7-06709ca417b7.png)

similar example with 保身に走る

![image](https://user-images.githubusercontent.com/35350819/113524592-2d089880-955c-11eb-8e08-dfa6f208654c.png)

![image](https://user-images.githubusercontent.com/35350819/113524589-2a0da800-955c-11eb-8649-e1831084e22a.png)

probably due to those dictionaries not having matching pos info

**Comment by toasted-nutbread** [comment]
Yes, it's due to the J2J dictionaries not having part of speech information, which is used to filter when inflections are involved. There is potentially a case to be made to not merge definitions into the same entry if they don't have matching POS info.

**Comment by Thermospore** [comment]
> There is potentially a case to be made to not merge definitions into the same entry if they don't have matching POS info.

Maybe they could still be pulled up, but with a tag to warn you the pos info is missing/doesn't match? Seems like the secondary search is a great way to get around dictionaries missing pos info. More pros than cons I think

**Comment by toasted-nutbread** [comment]
@FooSoft Should be ready for stable, updated the top comment.

**Comment by FooSoft** [comment]
Shipped!

**Comment by qharscoet** [comment]
@toasted-nutbread I guess the patch from #1591 didn't make it in the final release :'(

**Comment by toasted-nutbread** [comment]
Oh shoot, you're right, I forgot to include that in the list. I'll set up a patch release for @FooSoft.

https://github.com/FooSoft/yomichan/releases/tag/21.3.31.2

### General
* [ ] Create [release tag](https://github.com/FooSoft/yomichan/tree/21.3.31.2).
* [ ] Create [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.2).

### Testing
* [ ] Update Chrome testing release.
* [ ] Update Firefox testing release.
* [ ] Update Firefox [updates.json](https://github.com/FooSoft/foosoft.github.io/blob/master/projects/yomichan/dl/updates.json) and [verify](https://foosoft.net/projects/yomichan/dl/updates.json).
* [ ] Upload testing builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.2).

### Stable
* [ ] Update Chrome stable release.
* [ ] Update Firefox stable release.
* [ ] Upload stable builds to the [GitHub release](https://github.com/FooSoft/yomichan/releases/tag/21.3.31.2).

**Comment by FooSoft** [comment]
Shipped to testing and stable
